shardingsphere-mode/shardingsphere-mode-core/src/main/java/org/apache/shardingsphere/mode/manager/ContextManager.java

```diff
@@ -251,9 +251,7 @@ public void alterSchema(final String schemaName, final ShardingSphereSchema sche
      */
     public void alterSchema(final String schemaName, final TableMetaData changedTableMetaData, final String deletedTable) {
         ShardingSphereMetaData metaData = metaDataContexts.getMetaData(schemaName);
-        if (!containsInDataNodeContainedRule(changedTableMetaData.getName(), metaData)) {
-            metaData.getRuleMetaData().findRules(MutableDataNodeRule.class).forEach(each -> each.put(changedTableMetaData.getName(), each.getDataSourceNames().iterator().next()));
-        }
+        singleTableDataSourceMappingRelation(metaData, changedTableMetaData);
```

Comment:
Can we use another meaningful method name? Just like 'alterSingleTableDataNodes'?

Reply:
Yes, it' s a good idea.